api/rs/macros/lib.rs

```diff
@@ -345,6 +345,11 @@ fn extract_compiler_config(
                             _ => break,
                         }
                     }
+                    Some(TokenTree::Ident(debug_info_ident))
+                        if debug_info_ident.to_string() == "debug_info" =>
```

Comment:
Yes, this is sub-optimal and not quite correct. I did that just so that our own tests pass, but this is not really suitable. I'd keep this like the other options we have there, but they are undocumented.

Since we're not in build.rs but in the crate build itself, maybe `CARG_CFG_TEST` works - that would be nice. I can try that.

There's one other scenario and that's system tests for a Rust app that uses this macro. I ran into this in the example app, but I guess I need to change that to just not use the macro.

Reply:
Ah no, I think I got it wrong - we still need a solution for build.rs, and inside build.rs we don't know if later tests will be run or not.

Ok, alternate proposal:

Objective:

1) Support unit testing via search_api.rs for Rust applications, using build.rs and the slint! macro.
2) Support unit testing via slint-testing.h for C++ apps
3) Support system testing via Rust and a `slint/system-testing` feature
4) Support system testing via C++ and a `SLINT_FEATURE_SYSTEM_TESTING` feature

(In theory we have the same for JS and Python, but those use the interpreter and we always have the info available, or can access it if needed)

- We remove any `debug_info(bool)`/`--generate-debug-info=true` options from the compiler.
- We always emit the code to retrieve the element information, but we guard it with `#[cfg(test)]` and `#ifdef SLINT_ENABLE_DEBUG_INFO` (or so). This covers both (1) and (2).
- For (4), we could add a define to the interface definitions, and it'll be visible during application compilation.
- For (3) .... I don't know yet.... :-(